presidio-analyzer/presidio_analyzer/predefined_recognizers/fi_personal_identity_code_recognizer.py

```diff
@@ -0,0 +1,55 @@
+from datetime import datetime
+from typing import List, Optional
+
+from presidio_analyzer import Pattern, PatternRecognizer
+
+
+class FiPersonalIdentityCodeRecognizer(PatternRecognizer):
+    """
+    Recognizes and validates Finnish Personal Identity Codes (Henkilötunnus).
+
+    :param patterns: List of patterns to be used by this recognizer
+    :param context: List of context words to increase confidence in detection
+    :param supported_language: Language this recognizer supports
+    :param supported_entity: The entity this recognizer can detect
+    """
+
+    PATTERNS = [
+        Pattern(
+            "Finnish Personal Identity Code (Medium)",
```

Comment:
Another point, if ok- please add a reference to the logic (a government website if possible) for future users to see where this logic is coming from.

Reply:
@omri374 thanks. I have now made the changes.

Reply:
Thanks!